learn/components/styling.md

```diff
@@ -3,4 +3,309 @@ title: Styling
+</fancy-elements>
+```
+
+If you want to target elements in specific slots you can pass an attribute selector which matches the slot:
+
+```html
+<fancy-article>
+  <template shadowroot="open">
+    <style>
+      ::slotted([slot="title"]) {
+        font-size: 2rem;
+      }
+      ::slotted([slot="subtitle"]) {
+        font-size: 1.25rem;
+      }
+      /* The following will target elements going into the unnamed slot */
+      ::slotted(:not([slot])) {
+        color: deeppink;
+      }
+    </style>
+    <article>
+      <hgroup>
+        <slot name="title"></slot>
+        <slot name="subtitle"></slot>
+      </hgroup>
+      <slot></slot>
+    </article>
+  </template>
+  <h1 slot="title">I am the title</h1>
+  <p slot="subtitle">I am the subtitle</p>
+  <p>I am content.</p>
+</fancy-article>
+```
+
+You cannot chain selectors with `::slotted()`, so the following will not work:
+
+```css
+::slotted(h1) span {
+  color: deeppink;
+}
+```
+
+## Parts: styling a shadow tree from the outside
+
+The CSS Shadow Part API allows elements within a shadow tree to be styled from outside of it. This allows Web Components
+to be very extensible.
+
+```css
+fancy-article::part(header) {
+  display: grid;
+  gap: 0.25rem;
+  padding: 0.5rem;
+  border-block-end: 0.125rem solid deeppink;
+}
+
+fancy-article::part(content) {
+  display: grid;
+  justify-content: start;
+  gap: 0.5rem;
+}
+```
+
+```html
+<fancy-article>
+  <template shadowroot="open">
+    <article part="article">
+      <hgroup part="header">
+        <slot name="title"></slot>
+        <slot name="subtitle"></slot>
+      </hgroup>
+      <div part="content">
+        <slot name="content"></slot>
+      </div>
+    </article>
+  </template>
+  <h1 slot="title">Hello, World!</h1>
+  <p slot="subtitle">I am a subtitle...</p>
+  <p>I am content</p>
+</fancy-article>
+```
+
+Similar to `::slotted()`, you cannot chain selectors after `::part()` to select children or siblings. The following will
+not work:
+
+```css
+fancy-article::part(header) slot {
+  display: block;
+}
+```
+
+## How to include default styles for a Web Component
+
+There are a variety of methods to include styles for a Web Component. Some will be familiar, but others are newer.
+
+### Using `<style>`
+
+The `<style>` tag is the most simple way to write styles for a Web Component. Just include it in your shadow tree:
+
+```html
+<style>
+  :host {
+    color: deeppink;
+  }
+</style>
+```
+
+### Using `<link rel="stylesheet">`
+
+Using a `<link rel="stylesheet">` element in the shadow tree will allow you to write styles in an external stylesheet.
+
+```html
+<link rel="stylesheet" href="/fancy-article-element.css">
+```
+
+If you do this, the stylesheet will be loaded after the script is loaded. This will likely cause a “flash of unstyled
+content” (FOUC). To circumvent this, you can preload the stylesheet like this:
+
+```html
+<link rel="preload" href="/fancy-article-element.css" as="style">
+```
```

Comment:
This would need to be in `<head>` right?

Reply:
Not necessary, technically it could be anywhere since it isn’t rendered, but on page load `<head>` would be the best place. A scenario where it wouldn’t really matter if it were in the `<head>` would be if a web component was dynamically imported later on. That’s a more advanced use case though. The thing with this suggestion in general is that the preload and the usage of the component on the page can’t happen at different times, otherwise, the preload is worthless and you’ll get a console warning about it. I think maybe I can rework this suggestion to emphasize the caution in doing this. Generally I think `<style>` and constructable stylesheets seem better because of this.